inference-engine/src/inference_engine/ie_network_reader.cpp

```diff
@@ -151,6 +152,22 @@ void assertIfIRv7LikeModel(std::istream & modelStream) {
 }  // namespace
 
 CNNNetwork details::ReadNetwork(const std::string& modelPath, const std::string& binPath, const std::vector<IExtensionPtr>& exts) {
+    // Try to load with FrontEndManager
+    ngraph::frontend::FrontEndManager manager;
+    ngraph::frontend::FrontEnd::Ptr FE;
+    ngraph::frontend::InputModel::Ptr inputModel;
+    if (!binPath.empty()) {
+        FE = manager.load_by_model(modelPath, binPath);
```

Comment:
Why cannot we have check for `binPath.empty()` inside the frontend?

Reply:
That can be done, but it makes no sense to pass empty strings to frontend. Currently, it is treated as error.